### DISCUSSION TITLE
I want to debug the iteration of damask_grid to make it step by step

### DISCUSSION THREAD

**Discussion by ksy-fish** [damask-multiphysics/DAMASK#33] [Q&A] [answered]
I want to debug the iteration of damask_grid to make it step by step

I follow the usage in the website below
https://damask-multiphysics.org/documentation/usage.html#id3

In the section 2. Running the Simulation , I run the command below with the file in  [grid.tar.xz](https://damask-multiphysics.org/documentation/files/grid.tar.xz) on this page.
    `DAMASK_grid -l tensionX.yaml -g 20grains16x16x16.vti -m material.yaml `
This  command executed successfully.
I want to keep track of how the variables in the program change over the course of the iterations.
When i use gdb to debug DAMASK_grid,
    `gdb /usr/bin/DAMASK_grid 
`
it shows 
    ```
    Reading symbols from /usr/bin/DAMASK_grid...
    (No debugging symbols found in /usr/bin/DAMASK_grid)
    ```
I want to I hope to get your help, thanks.

**Comment by hyharry** [comment, marked as answer]
For that you need to build Damask yourself, i.e. `make BUILD_TYPE=debug`.

But prerequisite is that you need to know how to build petsc.

Pls take a look at
 https://damask-multiphysics.org/installation/source_code.html#source-code

**Reply by ksy-fish** [reply]
thank you,bro
